### PR TITLE
[inductor] Support mixed device in cpp wrapper

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #99950
* #99887
* #99203

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]

**Review by jansel — APPROVED** [review]

File: torch/_inductor/compile_fx.py [hunk 1]
```diff
@@ -525,14 +520,15 @@ def compile_fx_with_cpp_wrapper(
...
     """
-    from torch.ao.quantization.fx.utils import assert_and_get_unique_device
-
     # Turns off cpp_wrapper before calling back into compile_fx
     config_patches = {"cpp_wrapper": False}
-    device = assert_and_get_unique_device(module)
+    devices = (
+        {p.device.type for p in module.parameters()}
+        | {p.device.type for p in module.buffers()}
+        | {p.device.type for p in example_inputs}
```
Comment (chunyuan-w) [inline torch/_inductor/compile_fx.py:528]: In case `p` is a `SymInt`, the `p.device` call will fail here?

**Pull request closed without merging**